src/Homework1.java

```diff
@@ -1,101 +1,244 @@
 public class Homework1 {
 
     public static void main(String[] args) {
-        // Primitives
-        int myPrimitive = 10;
-        System.out.println(Integer.toBinaryString(myPrimitive));
-        System.out.println(Integer.toBinaryString(Integer.MAX_VALUE));
+        //Test1
+        System.out.println(sumUp(5));
+        //Test2
+        sumRange(1,100);
+        //Test3
+        fibonacci(20);
+        //Test4
+        System.out.println(firstAndLast(new int[] {13,6,2,8,2,3}));
+        System.out.println(firstAndLast(new int[] {13,6,2,8,2,13}));
+        //Test5
+        System.out.println(testArrayEquality(new int[] {1,2,3,4}, new int[] {1,2,3,4}));
+        System.out.println(testArrayEquality(new int[] {}, new int[] {}));
+        System.out.println(testArrayEquality(new int[] {1,3,2}, new int[] {1,3}));
+        System.out.println(testArrayEquality(new int[] {1,2,3,4}, new int[] {1,2,4,3}));
+        //Test6
+        concateneeteArrays(new int[] {1,2,3,4}, new int[] {5,6,7,8});
+        //Test7
+        minMaxAvg(new int[] {13,20,14,5,2,8});
+        //Test8
+        System.out.println(indexOfTarget(new int[] {13,20,14,5,2,8}, 5));
+        //Test9
+        oddEvenCounter(new int[] {13,20,14,5,2,8});
+        //Test10
+        System.out.println(testString("Hello java world", "java"));
+        //Test11
+        System.out.println(countAppearances("Hello java world", 'a'));
+        //Test12
+        System.out.println(endsInString("Hello java world", "orld"));
+        //Test13
+        System.out.println(alphabetIndex("abcd"));
+        //Test14
+        insertInString("Hello wooorld!");
+        //Test15
+        System.out.println(isPalindrome("aaabccbaaa"));
+        System.out.println(isPalindrome("abccbb"));
+    }
 
-        float myFloatPrimitive = 10.5f;
-        double myDoublePrimitive = 10.5f;
-        System.out.println(myFloatPrimitive);
+    //1
+    public static int sumUp(int number) {
+        int sum = 0;
+        for (int i = 1; i <= number; i++) {
+            sum += i;
+        }
+        return sum;
+    }
 
-        char myCharPrimitive = 'c';
+    //2
+    public static void sumRange(int first, int second) {
+        int sum = 0;
+        for (int i = first; i <= second; i++) {
+            sum += i;
+        }
+        System.out.println("Sum for range " + first + " to " + second + ": " + sum);
+        System.out.println("Average for range " + first + " to " + second + ": " + (sum/(first+second)));
```

Comment:
Media este suma numerelor pe cate sunt. Daca aduni finrst cu second nu o sa iti dea cate numere sunt intre first si second, ci doar suma lor. De exemplu suma de la 1 la 3 e 6 iar media lor este 2 (6/3). In cazul tau media e 6/4 (1+3=4)